lib/directory-monitor-poll.c

```diff
@@ -85,16 +109,31 @@ _rescan_directory(DirectoryMonitorPoll *self)
 {
   GError *error = NULL;
   GDir *directory = g_dir_open(self->super.real_path, 0, &error);
+  gchar *full_path;
+
   collection_comparator_start(self->comparator);
   if (directory)
     {
       const gchar *filename;
       while((filename = g_dir_read_name(directory)))
         {
-          collection_comparator_add_value(self->comparator, filename);
+          full_path = build_filename(self->super.real_path, filename);
+          collection_comparator_add_value(self->comparator, full_path);
+
+          /* compare entry mtime vs stat result */
+          CollectionComparatorEntry *entry = g_hash_table_lookup(self->comparator->original_map, filename);
+          if (entry->flag == IN_BOTH)
```

Comment:
@Kokan - FYI, I've tried this with and without the ``IN_BOTH`` condition and still only worked while editing the file and doing ``:w!`` only in vim.